### ISSUE TITLE
Question on completion implenetation and incomplete_compound_word on the first line of the document bug(?)

### ISSUE THREAD

**Issue by naquad** [nagaozen/gedit-plugin-autocomplete#5]
Hi.

Currently i'm modifying AutoComplete (ofcourse, if you'll be interested i'll send you result of this quest :)). Basic ideas are next:
1. Add suggestion source plugins. So user will be able to define his own completion suggestions. I've got this idea after i've tried to make AutoComplete use snippets as source (see P.S.).
2. I didn't like the way popup window was implemented. Currently i've created popup using gtk.TreeView (heavy, but looks very nice). New popup is able to show icons and customized easily. (I've wrote this after trying to add source icon (dictionary, buffer, snippets) to popup).
3. Extended configuration + plugins configuration.

The problem i've encountered is reproduced like this: first line of the document is processed incorrectly because regexp BEGINNING_OF_COMPOUND_WORD_SYNTAX doesn't match, so incomplete_simple_word is ok, but incomplete_compound_word contains only one letter.

I would like to fix this, but i can't :( my knowledge of python and gedit are very limited.

Could you please take a look at this problem?
Thank you.

P.S. About snippets: in gedit 2.28.3 (latest stable atm), they're accessible via gedit.plugins module, snippets class.
To get snippets for some language you should do: 

gedit.plugins.snippets.Library._instance.get_snippets(language_id)

to expand snippet you need to call
view_object._snippets_controller.run_snippet()
(snippet trigger still must be inserted).
As i've been told at irc.gnome.org #gedit this may not work in next versions.

P.P.S. I've got debian unstable. Here python version is 2.5 and there's no json module, installing from package (python-json by http://www.json.org) doesn't help, because it has different interface.

**Comment by naquad** [comment]
dirty hack applied:

```
    if len(incomplete_compound_word) < MIN_CHARACTERS_BEFORE_AUTOCOMPLETE and len(incomplete_simple_word) >= MIN_CHARACTERS_BEFORE_AUTOCOMPLETE:
        incomplete_compound_word=incomplete_simple_word
```

**Comment by nagaozen** [comment]
Should be fixed in 0.11.0
